sycl/include/CL/sycl/group_algorithm.hpp

```diff
@@ -257,7 +334,7 @@ joint_reduce(Group g, Ptr first, Ptr last, T init, BinaryOperation binary_op) {
            std::is_same<decltype(binary_op(init, *first)), float>::value),
       "Result type of binary_op must match reduction accumulation type.");
 #ifdef __SYCL_DEVICE_ONLY__
-  T partial = sycl::known_identity_v<BinaryOperation, T>;
+  T partial = detail::identity_for_ga_op<T, BinaryOperation>();
```

Comment:
isn't it better to update sycl::known_identity to let it accept std::complex? Otherwise, there may be confusion when known_identity states it is unknown but functionality is supported.

Reply:
We are asking this question of @v-klochkov above.  My feeling is that known_identity is used by the reductions and other parts of SYCL where std::complex is not supported. Adding complex there may not be wise.  Furthermore, the identity for complex numbers is {0,0} only in the case of addition  - which is the only binary operation the group algorithms support.  But known_identity accepts all the binary operations - so having unsupported ones may change its API in ways that confuse it or needlessly increase its test surface.

Reply:
I was hoping v-klochkov would be able to comment on this, but he is not going to be available to comment any time soon. We should not wait on him.
I have expanded the comment on `identity_for_ga_op`  and left a //TODO that we should eliminate it once complex number are supported by the other reductions.
@vladimirlaz and @Pennycook  would this be enough for now?
@Pennycook do you want to open a ticket for complex support for the reductions? (I think they are the primary other caller  of known_identity right now).

Reply:
I agree that sycl::has_known_identity should return true for std::plus<std::complex>.
It allows to enable std::plus for complex for reductions for free.